Assets/Scripts/KeyLocation/KeyLocationController.cs

```diff
@@ -109,7 +116,8 @@ public IEnumerator BeginDestroyingProcess()
         foreach (GameObject player in players)
         {
             player.GetComponent<FirstPersonController>().keyLocationSpeedMod = 1; //reset speedmod in case a player should be slowed by the edge when the location is disabled.
-            player.GetComponent<FirstPersonController>().isPlayerInKeyLocZone = false;
+            player.GetComponent<PlayerManager>().isPlayerInKeyLocZone = false;
+            player.GetComponent<FirstPersonController>().isPlayerInKeyLocZone = true;
```

Comment:
I don't think we want to have 2 values, which mean the same thing in 2 separate components. Could we reuse them?

Reply:
and also why are they set differently, so for one it's true and for other - false.